plugins/ltac/taccoerce.ml

```diff
@@ -144,9 +144,22 @@ let coerce_to_constr_context v =
     out_gen (topwit wit_constr_context) v
   else raise (CannotCoerceTo "a term context")
 
+let is_intro_pattern v =
+  if has_type v (topwit wit_intropattern [@warning "-3"]) then
+    Some (out_gen (topwit wit_intropattern [@warning "-3"]) v).CAst.v
+  else
+  if has_type v (topwit wit_simple_intropattern) then
```

Comment:
If you share the same runtime representation you can get rid of this additional check.

Reply:
Is it possible to share the same runtime representation but different parsing rules?

Reply:
Yep, `has_type` is only looking at the runtime representation.